src/components/Comments/NewCommentForm.tsx

```diff
@@ -0,0 +1,129 @@
+import {
+  FC, FormEvent, useState, useEffect,
+} from 'react';
+import classNames from 'classnames';
+import { TextField } from './TextField';
+import { CommentData, Field, FieldIcon } from '../../types';
+import { useCommentsContext } from '../../hooks/useCommentsContext';
+import { useGlobalContext } from '../../hooks/useGlobalContext';
+
+export const NewCommentForm: FC = () => {
+  const { resetError } = useGlobalContext();
+  const { onAddNewComment, commentIsProcessing } = useCommentsContext();
+
+  const [userName, setUserName] = useState('');
+  const [userEmail, setUserEmail] = useState('');
+  const [userCommentText, setUserCommentText] = useState('');
+
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [isReseted, setIsReseted] = useState(false);
+
+  const onSubmit = async (event: FormEvent) => {
+    event.preventDefault();
+    setIsSubmitted(true);
+
+    if (!userName || !userEmail || !userCommentText) {
+      return;
+    }
+
+    const newComment: CommentData = {
+      name: userName,
+      email: userEmail,
+      body: userCommentText,
+    };
+
+    const commentIsCreated = await onAddNewComment(newComment);
+
+    if (commentIsCreated) {
+      setUserCommentText('');
+    }
```

Comment:
it's not necessary. Just add try catch and reset comment text after successful adding